novawallet/Modules/Notifications/Setup/NotificationsSetupInteractor.swift

```diff
@@ -0,0 +1,107 @@
+import UIKit
+
+final class NotificationsSetupInteractor {
+    weak var presenter: NotificationsSetupInteractorOutputProtocol?
+
+    let servicesFactory: Web3AlertsServicesFactoryProtocol
+    let chainRegistry: ChainRegistryProtocol
+    private var syncService: Web3AlertsSyncServiceProtocol?
+    private var pushNotificationsService: PushNotificationsServiceProtocol?
+
+    let selectedWallet: MetaAccountModel
+    private var chains: [ChainModel.Id: ChainModel] = [:]
+
+    init(
+        servicesFactory: Web3AlertsServicesFactoryProtocol,
+        selectedWallet: MetaAccountModel,
+        chainRegistry: ChainRegistryProtocol
+    ) {
+        self.servicesFactory = servicesFactory
+        self.selectedWallet = selectedWallet
+        self.chainRegistry = chainRegistry
+    }
+
+    private func provideStatus() {
+        guard let pushNotificationsService = pushNotificationsService else {
+            return
+        }
+        pushNotificationsService.status { [weak self] status in
+            DispatchQueue.main.async {
+                self?.presenter?.didRegister(notificationStatus: status)
+            }
+        }
+    }
+
+    private func registerPushNotifications() {
+        guard let pushNotificationsService = pushNotificationsService else {
+            return
+        }
+        pushNotificationsService.setup()
+        pushNotificationsService.register { [weak self] in
+            self?.provideStatus()
+        }
+    }
+
+    private func saveSettingsAndRegisterDevice() {
+        guard let syncService = syncService else {
+            return
+        }
+        let chainFormat = ChainFormat.substrate(UInt16(SNAddressType.polkadotMain.rawValue))
+        let chainSpecific = selectedWallet.chainAccounts.reduce(into: [Web3AlertWallet.ChainId: AccountAddress]()) {
+            if let chainFormat = chains[$1.chainId]?.chainFormat {
+                let address = try? $1.accountId.toAddress(using: chainFormat)
```

Comment:
We need to use format from the chain. So, I would probably subscribe to chain registry, get the chains, unsubscribe and then create the settings.